CHANGELOG.md

```diff
@@ -0,0 +1,18 @@
+# 0.32 (in development)
+
+New features:
+
+* ...
+
+Bug fixes:
+
+* ...
+
+Performance:
+
+* Conversion of ASCII-only Ruby strings to Java strings is now faster.
+
+Internal changes:
+
+* The launcher is now part of the TruffleRuby repository, rather than part of
+the GraalVM repository.
```

Comment:
I think the ChangeLog should be user-oriented. So I think things should be listed only if they affect users at least potentially, and clarify what's the user-visible change if not obvious.

Reply:
I'm not so sure - I think big internal changes are worth mentioning as well. For example your shared objects work - that's totally invisible to users, but worth talking about as a change. This one perhaps isn't the best example. Something like adding a dependency on Sulong would be a better example.

Reply:
Right, big changes that could have non-trivial side-effects seem worth mentioning, as well as changes that affect contributors to some degree.
I was thinking to try to explain how the change affects users, but in this case it just shouldn't at all. (Using that launcher internally is another task/issue, and that would affect arguments parsing)

Reply:
Perhaps we could lean towards listing lots in the change log, and then go back before a release and think about what really matters? I think a list of internal changes would also be useful for us trying to keep up with what's changed.

Reply:
I think usually we don't really know in advance which commit will be picked for the release so that might be hard. But since the ChangeLog should be read from its version in `master` we could fix it around the time of the release, indeed.

Reply:
I agree it should be user visible changes. The worst sorts of changelogs are those that basically list commits. Having said that, this is also listed under "Internal changes" and I don't have a problem with that.